packit_service/worker/handlers/koji.py

```diff
@@ -305,14 +305,20 @@ def run(self):
             KojiBuildState.canceled: BaseCommitStatus.error,
         }.get(self.koji_build_event.state)
 
+        logger.info(f"Build status in DB: {self.build.status}")
         if (
             new_commit_status
             and self.build.status
-            and self.build.status != KojiBuildState.building
+            and self.build.status
+            in (
+                BaseCommitStatus.failure.value,
+                BaseCommitStatus.error.value,
+                BaseCommitStatus.success.value,
+            )
```

Comment:
Thanks it lgtm. I have just one question, that is not related with your refactoring but it is just something that I don't understand... why shouldn't we update the status supposing the `new_commit_status` is different from `KojiBuildState.building`?

Reply:
the point was to not overwrite the "final" status if we already have that in the DB, but since we set the value corresponding to BaseCommitStatus and not the KojiBuildState, therefore I changed it here

Reply:
I was just wondering why don't have the "real" (latest) *final* state. I can't imagine why we shouldn't have the latest one 🤔

Reply:
I think this could happen e.g. if we process the events in a different order that they were sent.

Reply:
Oh wired but thanks a lot for the answer!